python/3d/auto_generated/materials/qgsphongtexturedmaterialsettings.sip.in

```diff
@@ -65,11 +72,11 @@ Returns shininess of the surface
     virtual QMap<QString, QString> toExportParameters() const;
 
 
-    QString diffuseTexturePath() const;
+    QVariant diffuseTexture() const;
```

Comment:
This is an api break -- can you leave the existing getters/setters, but deprecate them?

Reply:
This is indeed an api break but the header contains the following sentence: 

```
\warning This is not considered stable API, and may change in future QGIS releases. It is
 * exposed to the Python bindings as a tech preview only.
 ```

Reply:
That's our "get out of jail" card for extreme cases only though -- wherever possible we should still be "nice" to plugin developers and avoid breaking API

Reply:
I don't mind deprecate the existing function but it does look like we some reliable rules to break it or not. 

I remember on a previous PR, you specifically asked me to break the API of some 3D enum. At the time, your argument was: "no one is really using that API." I could use the same argument here: I really don't think that anyone is using that API.

Reply:
The API is now preserved and the previous functions are deprecated.